src/proj.rs

```diff
@@ -180,14 +245,171 @@ impl Proj {
-        // available to all contexts
-        let dctx: *mut PJ_CONTEXT = ptr::null_mut();
-        unsafe { proj_context_set_search_paths(dctx, newlength, paths_p.as_ptr()) }
+        unsafe { proj_context_set_search_paths(self.ctx(), newlength, paths_p.as_ptr()) }
         Ok(())
     }
 
-    /// Create a transformation object that is a pipeline between two known coordinate reference systems.
+    /// Enable or disable the local cache of grid chunks
+    ///
+    /// To avoid repeated network access, a local cache of downloaded chunks of grids is
+    /// implemented as SQLite3 database, cache.db, stored in the PROJ user writable directory.
+    /// This local caching is **enabled** by default.
+    /// The default maximum size of the cache is 300 MB, which is more than half of the total size
+    /// of grids available, at time of writing.
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn grid_cache_enable(&self, enable: bool) {
+        let enable = if enable { 1 } else { 0 };
+        let _ = unsafe { proj_grid_cache_set_enable(self.ctx(), enable) };
+    }
+
+    /// Set the URL endpoint to query for remote grids
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn set_url_endpoint(&self, endpoint: &str) -> Result<(), ProjError> {
+        let s = CString::new(endpoint)?;
+        unsafe { proj_context_set_url_endpoint(self.ctx(), s.as_ptr()) };
+        Ok(())
+    }
+}
+
+impl Info for Proj {
+    #[doc(hidden)]
+    fn ctx(&self) -> *mut PJ_CONTEXT {
+        self.ctx
+    }
+}
+
+enum Transformation {
+    Projection,
+    Conversion,
+}
+
+/// [Information](https://proj.org/development/reference/datatypes.html#c.PJ_INFO) about PROJ
+#[derive(Clone, Debug)]
+pub struct Projinfo {
+    pub major: i32,
+    pub minor: i32,
+    pub patch: i32,
+    pub release: String,
+    pub version: String,
+    pub searchpath: String,
+}
+
+/// A `PROJ` Context instance, used to create a transformation object.
+///
+/// Create a transformation object by calling `transform_new` or `transform_known_crs`.
+pub struct TransformBuilder {
+    ctx: *mut PJ_CONTEXT,
+}
+
+impl TransformBuilder {
+    /// Create a new `TransformBuilder`, allowing grid downloads and other customisation.
+    pub fn new() -> Self {
+        let ctx = unsafe { proj_context_create() };
+        TransformBuilder { ctx }
+    }
+
+    /// Try to create a coordinate transformation object
+    ///
+    /// **Note:** for projection operations, `definition` specifies
+    /// the **output** projection; input coordinates
+    /// are assumed to be geodetic in radians, unless an inverse projection is intended.
+    ///
+    /// For conversion operations, `definition` defines input, output, and
+    /// any intermediate steps that are required. See the `convert` example for more details.
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn transform(mut self, definition: &str) -> Option<Proj> {
```

Comment:
another naming nit which you can feel free to take or leave: `proj` or `build_proj`